src/python/pants/rules/core/filedeps.py

```diff
@@ -37,17 +39,18 @@ class Filedeps(Goal):
 
 
 @goal_rule
-def file_deps(
+async def file_deps(
   console: Console,
   options: FiledepsOptions,
   build_root: BuildRoot,
   transitive_hydrated_targets: TransitiveHydratedTargets,
 ) -> Filedeps:
   unique_rel_paths: Set[str] = set()
   for hydrated_target in transitive_hydrated_targets.closure:
-    adaptor = hydrated_target.adaptor
-    unique_rel_paths.add(hydrated_target.address.rel_path)
+    bfa = await Get[BuildFileAddress](Address, hydrated_target.address)
```

Comment:
This `await` inside the loop like this will be very, very slow... probably a blocker.